app/controllers/application_controller.rb

```diff
@@ -27,6 +29,19 @@ def redirect_when_logged_in
     end
   end
 
+  def require_valid_subscription
+    unless logged_in? && (current_user.valid_subscription? || current_user.admin?)
+      flash[:danger] = "Your subscription has been cancelled.  Please update your credit card details to resubscribe to myflix."
+      redirect_to account_details_path unless request.path == '/account_details'
```

Comment:
I would recommend using the named route here instead of hard coding the path. Makes it much easier if you change the url later.